### ISSUE TITLE
Allow setting ignorePatterns in eslintrc

### ISSUE THREAD

**Issue by rightaway** [eslint/eslint#10891]
Could an option be added to specify the ignorepatterns in the eslintrc file rather than only on the command line? It would be useful to be able to see the ignored files right in the same config file as the rest of the configuration.

The current options are either to specify it in an eslintignore file (creates an extra file even for just one pattern), on the command line (needs to be added to the command invocation everywhere it's called), or in package.json (needs to be added to every project using eslintrc even if all projects might have the same ignored files). I don't think any of these options are ideal.

eslintrc is already the right place to put configuration and can elegantly handle configuration overrides, so it's a good place to have an option of putting the ignorepatterns. For example if I want to ignore something across all projects, it goes in the global ~/.eslintrc, but if I want to ignore it for a particular project, it goes in the project specific eslintrc.

**Comment by ilyavolodin** [comment]
You can have multiple eslintrc configs that override and extend each other. However, we only support on eslintignore file. We've tried cascading eslintignore file at some point early on, and it didn't work, due to the fact that you are not limited to the current folder in ignore file, you could say ignore file in the folder above current, by which time it's too late to do so.
So, I don't think what you are asking for is possible.

**Comment by rightaway** [comment]
I think ignorepatterns could still work in eslintrc files. If the pattern appears in the root `~/.eslintrc` (user's home directory), then it can be any ignorepattern that can currently appear in an `.eslintignore` file, and would be interpreted as such.

But any ignorepatterns appearing in the other eslintrc files would be interpreted relative to that eslintrc file (so patterns that extend outside the current folder would have no effect). Couldn't this handle cascading or am I missing something?

As long as the behavior is documented it could be an ideal solution to ignoring files, which has many benefits compared to the current options I described above.

**Comment by ilyavolodin** [comment]
That suggestion would go against format of the ignorefile (which is based on .gitignore). Because there's no rule that says that ignore file shouldn't have a pattern like this: `../somefile.js`. As I mentioned before, we had a version of eslintignore that supported cascading (I think before version 1.0), but it was riddled with bugs and issues, and was eventually replaced with simple root ignore file. This change might also result in performance degradation, due to the fact if all of the files in the current folder are ignored, you still need to read and parse config file to figure out that there's nothing to lint.
And another point is, config files can use `extend` property to inherit from other config files, as well as shareable configs. If shareable config will include ignore patter for some reason, there's absolutely not way for user to figure out why their files is not being linted, other then to go to shareable config and read the source-code.
Overall, I'm 👎 on this idea. ESLint has been around for over 5 years now, and we've only heard this suggestion maybe twice, which tells me that this would bring a very significant complexity to the core, to satisfy very rare edge case.

**Comment by phiresky** [comment]
Just wanted to note that currently I have to do this to make eslint work for my project:

```sh
eslint -- $(git ls-files '*.ts' '*.tsx' '*.js' '*.jsx')
```

which is kinda annoying because it depends on git and doesn't work for Windows devs. Might still be useful to others.

Also @platinumazure in https://github.com/eslint/eslint/issues/9329 incorrectly notes that `
--ignore-path` can be used for .gitignore files, which only works for the simplest of repositories with a single .gitignore file in the root of the project (no cascading).
